### PR TITLE
feat: pass additional parameters to the login function

### LINKED ISSUE #145: 💡 [REQUEST] - Pass extra parameters to login()-function
### Summary

Some login parameters can only be set dynamically, or depends on user behavior.
To support this, the AuthProviders exposed "login"-function should allow for "extraAuthParams"

### Basic Example

```react
<button onClick={() => login('customLoginState', extraParams={someParam: "value"})}>Login w/state</button>
```

### Drawbacks

none

### Unresolved questions

_No response_

### Implementation PR

_No response_

### Reference Issues

_No response_

### PR DESCRIPTION
**Pull request by soofstad**
## What does this pull request change?

- Add an optional parameter to `login()` that allows sending auth parameters others than those defined in the static config object.

## Issues related to this change
fixes #145

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ interface IAuthContext {
   tokenData?: TTokenData
   // Function to trigger login. 
   // If you want to use 'state', you might want to set 'clearURL' configuration parameter to 'false'.
-  login: (state?: string) => void  
+  login: (state?: string, additionalParameters?: { [key: string]: string | boolean | number }) => void
```
Comment (sebastianvitterso) [inline README.md:84]: Considering how this type is repeated a lot of places, maybe it should be defined somewhere else, and just reused? DRY and all that.

**Review by sebastianvitterso — APPROVED** [review]
Looks good! 
Complexity is creeping now, though, so if we're gonna keep adding non-spec-based functionality, it might get a bit unmanagable after a while?
Oh, and one comment.

File: src/Types.ts [hunk 2]
```diff
@@ -42,14 +42,16 @@ export interface IAuthProvider {
...
-  login: (state?: string, additionalParameters?: { [key: string]: string | boolean | number }) => void
+  login: (state?: string, additionalParameters?: TPrimitiveRecord) => void
   error: string | null
   tokenData?: TTokenData
   idToken?: string
   idTokenData?: TTokenData
   loginInProgress: boolean
 }
 
+export type TPrimitiveRecord = { [key: string]: string | boolean | number }
```
Comment (sebastianvitterso) [inline src/Types.ts:53]: Good name!

**Review by sebastianvitterso — APPROVED** [review]

**Comment by soofstad** [comment]
> Looks good! Complexity is creeping now, though, so if we're gonna keep adding non-spec-based functionality, it might get a bit unmanagable after a while? Oh, and one comment.

Agreed, this one however is not defying spec. Just makes it so you can set parameters used in the request dynamically, instead of only statically in the configObject.

Also, I can live with some complexity, as long as it does not bleed into userspace, making it hard to use the library.

**Pull request merged**